scipy/special/tests/test_legendre.py

```diff
@@ -0,0 +1,1476 @@
+        assoc_legendre_p_1_1_jac_div_z(z, typ = typ) / 2
+
+def assoc_legendre_p_4_2_jac(z, *, typ, norm = False):
+    typ_sign = np.where(typ == 3, -1, 1)
+    fac = assoc_legendre_factor(4, 2, norm)
+
+    return 30 * typ_sign * fac * (4 - 7 * z * z) * z
+
+def assoc_legendre_p_4_3_jac(z, *, typ, norm = False):
+    typ_sign = np.where(typ == 3, -1, 1)
+    fac = assoc_legendre_factor(4, 3, norm)
+
+    return 105 * typ_sign * fac * ((5 - 4 * z * z) * z * z - 1) * \
+        assoc_legendre_p_1_1_jac_div_z(z, typ = typ)
+
+def assoc_legendre_p_4_4_jac(z, *, typ, norm = False):
+    fac = assoc_legendre_factor(4, 4, norm)
+
+    return 420 * fac * (z * z - 1) * z
+
+def assoc_legendre_p_4_m4_jac(z, *, typ, norm = False):
+    fac = assoc_legendre_factor(4, -4, norm)
+
+    return fac * (z * z - 1) * z / 96
+
+def assoc_legendre_p_4_m3_jac(z, *, typ, norm = False):
+    fac = assoc_legendre_factor(4, -3, norm)
+
+    return fac * ((4 * z * z - 5) * z * z + 1) * \
+        assoc_legendre_p_1_1_jac_div_z(z, typ = typ) / 48
+
+def assoc_legendre_p_4_m2_jac(z, *, typ, norm = False):
+    typ_sign = np.where(typ == 3, -1, 1)
+    fac = assoc_legendre_factor(4, -2, norm)
+
+    return typ_sign * fac * (4 - 7 * z * z) * z / 12
+
+def assoc_legendre_p_4_m1_jac(z, *, typ, norm = False):
+    typ_sign = np.where(typ == 3, -1, 1)
+    fac = assoc_legendre_factor(4, -1, norm)
+
+    return typ_sign * fac * ((27 - 28 * z * z) * z * z - 3) * \
+        assoc_legendre_p_1_1_jac_div_z(z, typ = typ) / 8
+
+def sph_legendre_factor(n, m):
+    return assoc_legendre_factor(n, m, norm = True) / np.sqrt(2 * np.pi)
+
+def sph_legendre_p_0_0(phi):
+    fac = sph_legendre_factor(0, 0)
+
+    return np.full_like(phi, fac)
+
+def sph_legendre_p_1_0(phi):
+    fac = sph_legendre_factor(1, 0)
+
+    return fac * np.cos(phi)
+
+def sph_legendre_p_1_1(phi):
+    fac = sph_legendre_factor(1, 1)
+
+    return -fac * np.abs(np.sin(phi))
+
+def sph_legendre_p_1_m1(phi):
+    fac = sph_legendre_factor(1, -1)
+
+    return fac * np.abs(np.sin(phi)) / 2
+
+def sph_legendre_p_2_0(phi):
+    fac = sph_legendre_factor(2, 0)
+
+    return fac * (3 * np.square(np.cos(phi)) - 1) / 2
+
+def sph_legendre_p_2_1(phi):
+    fac = sph_legendre_factor(2, 1)
+
+    return -3 * fac * np.abs(np.sin(phi)) * np.cos(phi)
+
+def sph_legendre_p_2_2(phi):
+    fac = sph_legendre_factor(2, 2)
+
+    return 3 * fac * (1 - np.square(np.cos(phi)))
+
+def sph_legendre_p_2_m2(phi):
+    fac = sph_legendre_factor(2, -2)
+
+    return fac * (1 - np.square(np.cos(phi))) / 8
+
+def sph_legendre_p_2_m1(phi):
+    fac = sph_legendre_factor(2, -1)
+
+    return fac * np.cos(phi) * np.abs(np.sin(phi)) / 2
+
+def sph_legendre_p_3_0(phi):
+    fac = sph_legendre_factor(3, 0)
+
+    return fac * (5 * np.square(np.cos(phi)) - 3) * \
+        np.cos(phi) / 2
+
+def sph_legendre_p_3_1(phi):
+    fac = sph_legendre_factor(3, 1)
+
+    return -3 * fac * (5 * np.square(np.cos(phi)) - 1) * \
+        np.abs(np.sin(phi)) / 2
```

Comment:
Totally optional style opinion: not a fan of the `\`'s everywhere.

```suggestion
    res = (-3 * fac * (5 * np.square(np.cos(phi)) - 1)
           * np.abs(np.sin(phi)) / 2)
    return res
```

Reply:
Out of curiousity, what's your preferred alternative? Just brackets `(...)`?

Reply:
Yeah, just brackets. From Pep8 again

```
The preferred way of wrapping long lines is by using Python’s implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.
```

Reply:
Fine, happy with that.